src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -185,6 +188,95 @@ namespace OpenMS
     return integral;
   }
 
+  PeakIntegrator::PeakShapeMetrics_ PeakIntegrator::calculatePeakShapeMetrics(
+    const MSChromatogram& chromatogram,
+    const double& left,
+    const double& right
+  ) const
+  {
+    // TODO remove the three following lines
+    // peak_height_ = 965356;
+    // peak_apex_rt_ = 2.7045;
+    // background_height_ = 723.5;
+    PeakShapeMetrics_ peakShapeMetrics;
+    peakShapeMetrics.points_across_baseline = 0;
+
+    const double start_intensity = std::max(chromatogram.RTBegin(left)->getIntensity() - background_height_, 0.0);
+    const double end_intensity = std::max((chromatogram.RTEnd(right)-1)->getIntensity() - background_height_, 0.0);
+    // TODO remove the following couts
+    std::cout << std::endl << "int left: " << chromatogram.RTBegin(left)->getIntensity() << std::endl;
+    std::cout << "int right: " << (chromatogram.RTEnd(right)-1)->getIntensity() << std::endl;
+    std::cout << "bg_height: " << background_height_ << std::endl;
+    std::cout << "start_intensity: " << start_intensity << std:: endl;
+    std::cout << "end_intensity: " << end_intensity << std::endl;
+
+    MSChromatogram::ConstIterator it_rtbegin_l = chromatogram.RTBegin(left);
+    MSChromatogram::ConstIterator it_rtend_apex = chromatogram.RTEnd(peak_apex_rt_);
+    MSChromatogram::ConstIterator it_rtend_r = chromatogram.RTEnd(right);
+
+    peakShapeMetrics.start_time_at_5 = findRTAtPoint(it_rtbegin_l, it_rtend_apex, 0.05, true);
```

Comment:
I like the encapsulation of this into a function.  However, it is quite a few calls that iterate over the peak.  Have you timed the code like this versus what I had previous?  I am curious to see if there is any time penalty.

Reply:
My version appears to be slower (something like 0.000002 against 0.000004 seconds)
I'm guessing the time is spent on the `find_if()`, but can't be sure without profiling (and with such small inputs I guess it wouldn't make much sense to over-optimize)